src/lib/image/squashfs/init.c

```diff
@@ -76,8 +93,41 @@ int _singularity_image_squashfs_init(struct image_object *image, int open_flags)
     /* if LAUNCH_STRING is present, figure out squashfs magic offset */
     p = strstr(buf, "hsqs");
     if ( p != NULL ) {
-        singularity_message(VERBOSE2, "File is a valid SquashFS image\n");
+        char *compression_type = NULL;
+
         image->offset = p - buf;
+
+        if ( ( image->offset + sizeof(struct squashfs_header) ) > sizeof(buf) ) {
+            singularity_message(VERBOSE, "File is not a valid SquashFS image\n");
+            return(-1);
+        }
+
+        singularity_message(VERBOSE2, "File is a valid SquashFS image\n");
+
+        s_header = (struct squashfs_header *)p;
+
+        switch( s_header->compression ) {
+        case SQUASHFS_LZMA:
+            compression_type = strdup("lzma");
+            break;
+        case SQUASHFS_LZO:
+            compression_type = strdup("lzo");
+            break;
+        case SQUASHFS_XZ:
+            compression_type = strdup("xz");
+            break;
+        case SQUASHFS_LZ4:
+            compression_type = strdup("lz4");
+            break;
+        default:
+            break;
+        }
+
+        if ( compression_type ) {
```

Comment:
I'm not sure it's necessary to spit out a warning every time an image is run that was compressed. I think it makes more sense to output this to DEBUG.

Reply:
Was discussed on slack to display a warning because that would be source of confusion for users downloading image not compressed with gzip by default, some kernel don't support all squashfs compression format by default, then display a warning here make sense to warn users about potential problem